src/textual/widget.py

```diff
@@ -2743,7 +2743,12 @@ def watch_has_focus(self, value: bool) -> None:
 
     def watch_disabled(self) -> None:
         """Update the styles of the widget and its children when disabled is toggled."""
-        self.blur()
+        if (
+            self.disabled
+            and self.app.focused is not None
+            and self.app.focused in self.walk_children()
```

Comment:
It is somewhat inefficient to walk the DOM every time something is disabled.

How about we just check if `self.app.focused` is disabled or has a disabled ancestor?

Reply:
🤔 as in a45b156?

Reply:
Yeah, that looks right to me.

Reply:
But the tests don't like it. Why does this require a screen on the stack..?

Reply:
@willmcgugan I'd say you have to check if the optimisation is worth the extra mess created to handle the cases where the `disabled` status is set outside the context of a running app.